### ISSUE TITLE
[FN]: $variable seems to not be defined in scope.

### ISSUE THREAD

**Issue by j-applese3d** [kalessil/phpinspectionsea#1898]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | com.kalessil.phpStorm.phpInspectionsEA (5.0.0.0) |
| Language level | e.g. PHP 7.2 (check project settings or composer.json)                        |

Current behaviour
--
```php
function hi($in): mixed {
    
    $x = $good ?? [];
    $x2 = [$bad1] ?? [];
    $x3 = hi([$bad2]) ?? [];
    $x4 = hi([$bad3]) ?? '';
    
    return $_REQUEST['a'];
}
```
![image](https://github.com/kalessil/phpinspectionsea/assets/24463871/07fbb542-7d32-426c-9cb6-811632c23fde)

Expected behaviour
--
It correctly handles `$good` by highlighting it as a probable issue.
However, when the undefined variable is tucked within an array, it doesn't seem to care?
Every `$bad#` variable should be highlighted.

Environment details
---
PhpStorm 2023.1.3
Build #PS-231.9161.47, built on June 21, 2023
Runtime version: 17.0.7+10-b829.16 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 5.19.0-46-generic

**Comment by uncaught** [comment]
Is this the same problem?

This gets reported as bad:

![image](https://github.com/user-attachments/assets/52d27fbb-3206-4f76-afc0-f58a29679fe5)

But this does not:

![image](https://github.com/user-attachments/assets/a689944e-430f-4751-88ab-5958284a2152)

I would like to have an inspection error for _every_ variable, which is not a known global, if it is not in scope.
